### ISSUE TITLE
Allow to set recovery_target_time in the pitr init mode

### ISSUE THREAD

**Issue by trojanowski** [sorintlab/stolon#296]
I'd like to have an option to restore cluster to a specific point in time, e. g. by running:

```
stolonctl init '{ "initMode": "pitr", "pitrConfig":  {"recoveryTargetSettings": {"recoveryTargetTime": "..."}}}'
```

**Comment by sgotti** [comment]
@trojanowski I created an initial pull request #303 

if you can test it and report if it works I'll be very happy.

**Comment by trojanowski** [comment]
@sgotti I tried it and noticed some problems.

When I used only `recoveryTargetTime` in `recoveryTargetSettings` I had the following error:

```
FATAL:  invalid value for recovery parameter "recovery_target": ""
HINT:  The only allowed value is "immediate".
```

When I also added `"recoveryTarget": "immediate"` the error was different:

```
.FATAL:  unrecognized recovery parameter "recovery_target_lsn"
```

I was using Postgres 9.6 and `recovery_target_lsn` seems to be supported since version 10.0.

**Comment by trojanowski** [comment]
I think a solution could be to set in _recovery.conf_ only options which were explicitly defined in the `stolonctl init` command. Currently a sample generated  _recovery.conf_ file looks like the one below:

```
recovery_target_name = ''
recovery_target_time = '2017-07-05 13:45:00'
recovery_target_xid = ''
recovery_target_timeline = ''
restore_command = 'wal-e wal-fetch "%f" "%p"'
recovery_target = 'immediate'
recovery_target_lsn = ''
```

**Comment by sgotti** [comment]
@trojanowski yeah, forgot to check if they are empty before setting them, will update tomorrow.

**Comment by sgotti** [comment]
@trojanowski updated PR

**Comment by sgotti** [comment]
@trojanowski merged #303. I'm closing this. If you manage to test it and find some issues please let us know. Thanks!
